sklearn/metrics/classification.py

```diff
@@ -239,6 +242,11 @@ def confusion_matrix(y_true, y_pred, labels=None):
     else:
         labels = np.asarray(labels)
 
+    if sample_weight is None:
+        sample_weight = np.ones(y_true.shape[0], dtype=np.int)
+    else:
+        sample_weight = np.asarray(sample_weight)
+
```

Comment:
@MechCoder sure, seems like a good addition. I added the check after the highlighted code just to make sure that in the future, if some part of the previous code changes, the check still catches any future problems.

It would be good to have some tests against this behavior, but I'm a little short on time in the following weeks, so it's probably better to merge this and later I'll add another PR to improve the tests. What do you think?